lists2webkit.py

```diff
@@ -0,0 +1,124 @@
+import json
+import os
+from settings import config
+from constants import (
+    DNT_SECTIONS,
+    PRE_DNT_SECTIONS,
+    ENTITYLIST_SECTIONS,
+    WEBKIT_LISTS_DIR,
+    WEBKIT_BLOCK_ALL,
+    WEBKIT_BLOCK_COOKIES
+)
+from utils import (
+    get_blocked_domains,
+    add_domain_to_list,
+    load_json_from_url
+)
+
+def process_and_sort_domains(domains):
+    """Sorts and adds domains to the list, returning successfully added domains."""
+    added_domains = []
+    previous_domain = None
+    output = []
+    for domain in sorted(domains):
+        if add_domain_to_list(domain, domain, previous_domain, None, output):
+            added_domains.append(domain)
+            previous_domain = domain
+    return added_domains
+
+def get_tracker_lists(section):
+    """Retrieves and processes tracker lists for a given section."""
+    blocked_domains = get_blocked_domains(config, section)
+    processed_domains = process_and_sort_domains(blocked_domains)
+    print(f"Processing tracker list: {section}")
+    return {section: processed_domains}
+
+def get_entity_list(section):
+    """Retrieves the entity whitelist for a given section."""
+    entity_data = load_json_from_url(config, section, 'entity_url')
+    list_name = config.get(section, 'output')
+    print(f"Processing entity list: {list_name}")
+    return {
+        res: {"properties": details.get("properties", [])}
+        for details in entity_data.get("entities", {}).values()
+        for res in details.get("resources", [])
+    }
+
+def build_url_filter(resource):
+    escaped_resource = resource.replace('.', '\\.')
+    return f"^https?://([^/]+\\.)?{escaped_resource}"
+
+def find_entity_for_resource(resource, entities):
+    for key_resource, entity in entities.items():
+        if key_resource in resource:
+            return entity
+    return None
+
+def build_rule(resource, action_type, entities):
```

Comment:
Could you add a comment to explain how we build webkit rules based on the blocking domains and entities? It would be good for people to understand.

Reply:
Done !